addons/stock_account/views/stock_account_views.xml

```diff
@@ -30,10 +30,12 @@
             <field name="inherit_id" ref="stock.view_location_form"/>
             <field name="arch" type="xml">
                 <xpath expr="//group[@name='additional_info']" position="after">
-                    <group string="Accounting Information" attrs="{'invisible':[('usage','not in',('inventory','production'))]}">
-                        <field name="valuation_in_account_id" options="{'no_create': True}"/>
-                        <field name="valuation_out_account_id" options="{'no_create': True}"/>
-                    </group>
+                    <div groups="stock_account.group_stock_accounting_automatic">
+                        <group string="Accounting Information" attrs="{'invisible':[('usage','not in',('inventory','production'))]}">
```

Comment:
Price difference account should also be hidden under automatic accounting since it's never used without it. To verify